test/run_tests.sh

```diff
@@ -0,0 +1,259 @@
+#!/bin/bash
+
+export TRY_TOP=${TRY_TOP:-$(git rev-parse --show-toplevel --show-superproject-working-tree)}
+export WORKING_DIR="$TRY_TOP/test"
+export RESOURCE_DIR="$WORKING_DIR/resources"
+export MISC_SCRIPT_DIR="$WORKING_DIR/misc"
+
+echo "=================| Try Tests |==================="
+echo "Test directory:           $WORKING_DIR"
+echo "Resource directory:       $RESOURCE_DIR"
+
+## Set the DEBUG env variable to see detailed output
+DEBUG=${DEBUG:-0}
+
+
+bash="bash"
+try="$TRY_TOP/try"
+
+try_workspace="$TRY_TOP/test/try_workspace"
+bash_workspace="$TRY_TOP/test/bash_workspace"
+
+# Results saved here
+output_dir="$TRY_TOP/test/results"
+
+echo "Bash test workspace:      $bash_workspace"
+echo "try test workspace:       $try_workspace"
+echo "Results saved at:         $output_dir"
+echo "================================================="
+
+# Clear previous test results
+rm -rf "$output_dir"
+mkdir -p "$output_dir"
+touch "$output_dir/result_status"
+
+cleanup()
+{
+    rm -rf "$try_workspace"
+    mkdir "$try_workspace"
+    rm -rf "$bash_workspace"
+    mkdir "$bash_workspace"
+}
+
+run_test()
+{
+    cleanup
+    local test=$1
+
+    if [ "$(type -t $test)" != "function" ]; then
+        echo "$test is not a function!   FAIL"
+        return 1
+    fi
+
+    echo -n "Running $test..."
+
+    # cd $try_workspace
+    $test "$bash" "$bash_workspace"
+    test_bash_ec=$?
+    # Run test
+    $test "$try" "$try_workspace"
+    test_try_ec=$?
+    # Check test EC
+    
+    diff -q "$bash_workspace/" "$try_workspace/" > /dev/null
+    test_diff_ec=$?
+    if [ $test_diff_ec -ne 0 ]; then
+        echo -n " (!) output mismatch"
+        echo "$test are not identical" >> $output_dir/result_status
+        echo -e '\t\tFAIL'
+    else
+        if [ $test_bash_ec != $test_try_ec ]; then
+            echo -n " (!) EC mismatch"
+            echo "$test are not identical" >> $output_dir/result_status
+            echo -e '\t\tFAIL'
+        else
+            echo -ne '\t\t\t'
+            echo "$test are identical" >> $output_dir/result_status
+            echo -e '\tOK'
+        fi
+    fi
+}
+
+test_untar_no_flag()
+{
+    local shell=$1
+    cp $RESOURCE_DIR/* "$2/"
+    # Will always commit the result in case of try
+    if [ $shell == "bash" ]; then
```

Comment:
"$(basename shell)"? [https://github.com/gliargovas/try/actions/runs/5176411081/jobs/9325190327#step:3:62] getting a syntax error...

Reply:
Fixed it with badd996